### DISCUSSION TITLE
Akash Grant Application: Open-Source Native Staking Tool

### DISCUSSION THREAD

**Discussion by smokingfive** [akash-network/community#700] [Integrations]
# Akash Grant Application: Open-Source Native Staking Tool

**Company**: Colossus Digital  
**Date**: October 2024

### Company Summary
At Colossus, we represent a fusion of dedication and innovation in the blockchain realm. Our services involve the integration of FIAT and crypto-regulated custody, complemented by our multi-chain staking and streamlined on/off-ramp solutions. Guided by the oversight of the Italian OAM, we've established a framework that underscores security, transparency, and client-centricity.

Our vision is clear: to enable institutional functions within the web3 environment. As blockchain technology expands across industries, we've developed a robust platform for efficient and secure transaction validation.

### Project Overview
We propose to build an open-source platform that enables the Akash developer community to transfer, delegate, undelegate, re-delegate tokens, withdraw rewards or commissions, and leverage REStake auto-compound capabilities ([REStake](https://restake.app/akash)) from the Akash network using their chosen custodian. This tool will allow users to interact with the Akash blockchain without needing to move assets to a custodian supported directly by Akash.

### Project Team
- **Development Team**  
  - Senior developers with extensive experience in Rust, Python, Javascript, Typescript, Blockchain development, and decentralized infrastructure.
  - Proven track record of developing crypto projects, focusing on scalability, security, and user accessibility.
- **Marketing Team**  
  - Experienced professionals in the blockchain and technology sectors, specializing in growth strategies, community engagement, and public relations. The team will promote the tool within the Akash community and the broader crypto ecosystem.

## Key Functionalities
1. **Transaction Crafting**  
   - Users can create transaction requests for transfer, delegating, undelegating, re-delegating, withdrawing rewards or commissions, and setting/unsetting REStake auto compound ([REStake](https://restake.app/akash)) on the Akash platform.
   
2. **Custodian Integration**  
   - Users can select their preferred custodian to handle assets, and transaction requests will be sent to the custodian for signing.  
   - Signed transactions ensure compliance with custodial policies.

3. **Broadcasting Transactions**  
   - Once the transaction is signed by the custodian, the user can submit it to the Akash network for execution.

4. **Open-Source Environment**  
   - Open-source code, APIs, deployment procedures, and infrastructure setup will be available on GitHub for community review and collaboration.

## Technical Details
- **API Accessibility**  
  - APIs will allow developers to generate supported transaction requests, retrieve transaction digests, and submit signed transactions to the Akash network.

- **Custody Signing Mechanism**  
  - Secure transmission of transaction digests to custodians for signing.

- **Deployment Procedures**  
  - A comprehensive guide covering installation, configuration, and execution will be provided for easy adoption.

## Impact on the Akash Ecosystem
- **Increased Participation**  
  - The tool will encourage more users, especially institutional participants, to delegate and undelegate tokens on the Akash network.

- **Enhanced Security**  
  - By allowing users to manage assets through their preferred custodians, the tool reduces risks associated with moving assets between platforms.

- **Open Source Contribution**  
  - Contributions to the growing library of open-source tools in the Akash ecosystem will foster innovation and collaboration.

## Timeline and Budget
- **Phase 1**: Design and API Development (Week 1-5)  
  - 45% of the total budget ($22,500)

- **Phase 2**: Integration with Custodians (Week 5-7)  
  - 25% of the total budget ($12,500)

- **Phase 3**: Open-Source Deployment and Documentation (Week 8-9)  
  - 15% of the total budget ($7,500)

- **Phase 4**: Testing and Community Feedback (Week 10-11)  
  - 15% of the total budget ($7,500)

Total Funding Request: $50,000

## Resource Breakdown
- **Blockchain Expert Leader**: 208 hours over 11 weeks  
- **DevOps Expert**: 104 hours over 11 weeks  
- **Backend Developers**: 288 hours over 11 weeks  
- **Project Manager**: 20 hours over 11 weeks  
- **Organisational Support**: 20 hours over 11 weeks  

## Milestones
1. **Milestone 1**: 70% of API Development  
   - Expected Payment: 30% of the total budget.

2. **Milestone 2**: 100% API Development and Custodian Integration  
   - Expected Payment: 40% of the total budget.

3. **Milestone 3**: 100% Open-Source Deployment, Documentation, and Community Feedback  
   - Expected Payment: 30% of the total budget.

## Conclusion
By providing a tool that allows the Akash community to securely delegate and undelegate tokens using custodians, we aim to enhance participation on the network, promote secure asset management, and contribute to the growth of the Akash ecosystem. This project aligns with Akash’s mission to provide decentralized cloud solutions with maximum flexibility and security for users.

## Contact Information
- Raj Panesar: [raj@colossus.digital](mailto:raj@colossus.digital)  
- Lorenzo Barbantini: [lorenzo@colossus.digital](mailto:lorenzo@colossus.digital)

[Colossus Digital](https://colossus.digital)  
Via del Commercio, 32, 00154 Roma  
P.IVA IT16238821009

**Comment by rodri-r** [comment]
Thank you for the information, however there are already several dashboards/explorers/wallets that already do everything that you plan to offer in your staking tool, one of them you even mention in your application -  https://restake.app/ which offers exactly the same and it is open source too.  

There is also Keplr wallet and dashboard, Leap wallet, Cosmostation wallet, Ping Explorer and others which are also open source. 

What differences are there between all these and your staking tool?

Also on these tools there are no "custodians" as you mention.  Each user has their tokens in their own wallet and can delegate/undelegate or switch validators and claim rewards at any time. Validators never have access to user keys or wallet.

**Comment by smokingfive** [comment]
## Differences Between Colossus and Other Staking Tools

Thank you for the feedback! We appreciate the opportunity to clarify the differences between Colossus and other tools like Restake, Keplr, Leap, Cosmostation, and Ping Explorer. While these platforms offer great solutions for retail users, Colossus is specifically designed to meet the needs of **institutions**. Below are the key distinctions:

### 1. Institutional Focus

Colossus is positioned as the **Keplr, Leap, or Cosmostation equivalent** for **institutions**. It provides a secure layer that bridges institutional clients and their custodians. Institutions have strict requirements around compliance, security, and governance. Colossus facilitates seamless interaction between institutions and custodians, enabling institutions to **earn a yield on their assets** while meeting regulatory and operational requirements.

### 2. Custodial Integration

Unlike the retail-focused tools, which emphasize **self-custody**, Colossus integrates directly with **institutional custodians** such as:

- **Fireblocks**
- **Ledger Enterprise**
- **Dfns**

Institutions typically don’t hold assets directly in wallets like retail users. Instead, they use custodial services to manage large amounts of assets securely. Colossus acts as the interface for institutions to engage in staking activities **without needing direct control over the assets themselves**, ensuring the highest levels of security and compliance.

### 3. Multi-Blockchain Support

Colossus supports a wide range of blockchains, allowing institutions to stake assets across various networks, all from a **single, unified operational framework**. This scalability is required for institutional clients who often manage **large portfolios across multiple chains** and need a solution that simplifies cross-chain asset management.

### 4. Existing Adoption

Colossus is already trusted by **several exchanges, institutions, and family offices** for staking and yield generation. This highlights our track record in the institutional space. Moving forward, we would like to extend this functionality to the **Akash Foundation**, opening up these institutional-grade staking services to a broader audience.

---

### Summary

While retail-focused tools like Restake, Keplr, Leap, and Cosmostation prioritize self-custody for individual users, **Colossus is designed for institutions**, addressing the complex, secure, and compliant needs of institutional players. By integrating with custodians and offering multi-chain support, Colossus provides a distinct, institutionally-focused solution for yield generation and staking.

**Comment by rodri-r** [comment]
Thank you for the explanation. 

In my opinion, I don't see much use for Colossus.  Institutions can already securely self-custody their assets with Keplr, Leap, Cosmostation, Ping, Ledger and other similar tools. all with multi chain support.  Institutions can already freely participate in governance, staking, compounding,etc. There are already some multi-sig tools available as well, which increase security, as well as permissions and allowances. There are also several tools that help both individuals and institutions to calculate their taxes easily.  

Is Colossus already working with some institutions?
If so, can you name a few please?

**Reply by smokingfive** [reply]
Thank you for your opinion. We understand your perspective, but Colossus is addressing **specific institutional needs** that go beyond retail tools. Here's why:

### Regulatory and Compliance Constraints

While retail users can **self-custody** assets using tools like Keplr, Ledger, or multi-sig wallets, **institutions are unable to self-custody** due to strict regulatory and compliance requirements. This is exactly why custodial services like:

- **Fireblocks**
- **Ledger Enterprise**
- **Dfns**

exist and are widely used in the institutional space. They provide the **necessary custodial framework** to ensure institutional assets are managed in a compliant manner. Colossus **adds value** by integrating with these custodians and **unlocking the potential** for institutions to stake, earn yield, and participate in governance **while remaining compliant** with regulatory standards.

### Unlocking Assets for Institutions

Colossus extends the institutional portfolio by enabling institutions to earn yield on their assets **securely and compliantly** through staking. While self-custody and multi-sig solutions are great for individuals and smaller entities, **larger institutions** require a layer of protection, compliance, and reporting that Colossus facilitates by integrating directly with custodial services.

### Current Institutional Adoption

Yes, Colossus is already working with several institutions. However, due to **non-disclosure agreements (NDAs)**, we are currently unable to disclose specific names. What we can share is that our platform is actively being used by **exchanges, institutions, and family offices**, and  our approach to the **blockchains with the most institutional demand**.

While we're not the only validator in this space, Colossus is **dynamic in our approach**, and we continue to open our platform to the blockchains that are most relevant to institutional players. We'd like to encourage you to explore further, as several institutions are already seeing the benefits of using these services for their staking needs.
